test/passes/dce_all-features.wast

```diff
@@ -810,4 +810,50 @@
       )
     )
   )
+  (func $unnecessary-concrete-block (result i32)
+    (block $foo (result i32) ;; unnecessary type
+      (nop)
+      (unreachable)
+    )
+  )
+  (func $necessary-concrete-block (result i32)
+    (block $foo (result i32)
+      (br $foo (i32.const 1))
+      (unreachable)
+    )
+  )
+  (func $unnecessary-concrete-if (result i32)
+    (if (result i32) ;; unnecessary type
+      (i32.const 0)
+      (return (i32.const 1))
+      (unreachable)
+    )
+  )
+  (func $note-loss-of-if-children
+    (block $label$1
+     (if ;; begins unreachable - type never changes - but after the condition
+         ;; becomes unreachable, it will lose the children, which means no more
+         ;; br to the outer block, changing that type.
```

Comment:
Not sure what changes in the outer block... Doesn't the outer block also begins unreachable, given that it does not have a type attached and its only child (`if`) starts unreachable?

Reply:
No, the block has a `br` to it (showing it can be reached and exited normally), so it begins with type `none`. Same as

```wat
(block $foo ;; type none
  (block $child ;; type unreachable
    (br $foo) ;; this makes $foo reachable
    (unreachable)
  )
)
```

Reply:
I see, thanks. (You mean, in the comment in your code snippet, "this makes $child unreachable"?)

Reply:
Oh, it does both I guess, yes: that one `br` makes `child` unreachable, and also makes `foo` reachable.